util/network-devp2p/src/connection.rs

```diff
@@ -317,14 +315,13 @@ impl EncryptedConnection {
 		let key_material_keccak = keccak(&key_material);
 		(&mut key_material[32..64]).copy_from_slice(key_material_keccak.as_bytes());
 
-		let iv = vec![0u8; 16];
-		let encoder = CtrMode::new(AesSafe256Encryptor::new(&key_material[32..64]), iv);
-		let iv = vec![0u8; 16];
-		let decoder = CtrMode::new(AesSafe256Encryptor::new(&key_material[32..64]), iv);
-
-		let key_material_keccak = keccak(&key_material);
-		(&mut key_material[32..64]).copy_from_slice(key_material_keccak.as_bytes());
-		let mac_encoder = EcbEncryptor::new(AesSafe256Encryptor::new(&key_material[32..64]), NoPadding);
+		// TODO: clarify this: ecdh::agree creates a **NEW** secret right? And AesCtr256 keeps an internal counter, right?
+		// Using a 0 IV with CTR is fine as long as the same IV is never reused with the same key. This is not the case here.
```

Comment:
if ecdh creates a new secret which will be the symetric key used only for this session the 0 iv is only use once with this secret, so we are in the case of same iv use for different key.